### ISSUE TITLE
Style UPE element Hover states

### ISSUE THREAD

**Issue by bborman22** [Automattic/woocommerce-payments#1829]
The goal of this issue is to apply the store theme's styles for element states (focus, hover, etc) to the UPE elements. This is broken out into a separate task as it is not as trivial and requires some spoofing of elements to get the appropriate styles. Inspiration from this issue can be found in [this proof of concept](https://github.com/Automattic/woocommerce-payments/pull/1765/files). There is also some [further discussion](paJDYF-1vP-p2#comment-5997) on this topic that could address some of the goals / concerns with this task.

**Comment by bborman22** [comment]
It would make sense to add this in after #1827 is complete.

**Comment by brezocordero** [comment]
I am separating the original issue into 2, keeping this one for Hover styles and creating a separate one for Focus and Error states.

Hover states are not computed from the page. Hover states for tabs are generated following these rules:

* Tabs on hover will have a darker background than current background.
* If current background color has brightness under 50 (Storefront is 51), the tab on Hover becomes lighter.
* Color for labels or fillable SVGs do not change on Hover. If the combination new background color/ text color is not readable, we use a fallback of black or white, depending on the background color. (This is handled using Tinycolor library)

Code is on branch [upe/1829-UPE-hover-styling](https://github.com/Automattic/woocommerce-payments/tree/upe/1829-UPE-hover-styling) waiting for #1826 and #1827 to be merged.
